### DISCUSSION TITLE
Line numbers for non-JSON formats

### DISCUSSION THREAD

**Discussion by prescod** [pydantic/pydantic#4254] [Pydantic V2]
I understand that you might not want to support direct parsing of N other alternatives to JSON, but please make your parser/line-number-reporting infrastructure generic so that any parser can be plugged in or use Pydantic downstream and have a way of reporting line numbers.

The lack of this support is the primary reason that I do not use Pydantic in one of my projects for Salesforce, which defines a YAML file format.

Ideally there would be third-party parsers that bring Pydantic's support for other languages to the same level as JSON: Pydantic-YAML, Pydantic-TOML, Pydantic-INI, etc.

The very least I need, though, is line number support.

**Comment by samuelcolvin** [comment]
Ye of course, if we add it, it'll be generic, in fact if you need it, we could probably add this before working on JSON line numbers.

As it happens, I'm working on a PR now on pydantic-core https://github.com/samuelcolvin/pydantic-core/pull/185 which could quite easily add support for line numbers.

Basically when creating a `PydanticValueError` (they way in pydantic V2 to raise an exception with custom details) you would just specify `..., file_position=(123, 4))`. Would this work for you?

**Reply by samuelcolvin** [reply]
https://github.com/samuelcolvin/pydantic-core/issues/187
